gocd-server/alpine-start.sh

```diff
@@ -0,0 +1,136 @@
+}
+
+die() {
+    yell "$1"
+    exit ${2:-1}
+}
+
+autoDetectJavaExecutable() {
+  local java_cmd
+  # Prefer using GO_JAVA_HOME, over JAVA_HOME
+  GO_JAVA_HOME=${GO_JAVA_HOME:-"$JAVA_HOME"}
+
+  if [ -n "$GO_JAVA_HOME" ] ; then
+      if [ -x "$GO_JAVA_HOME/jre/sh/java" ] ; then
+          # IBM's JDK on AIX uses strange locations for the executables
+          java_cmd="$GO_JAVA_HOME/jre/sh/java"
+      else
+          java_cmd="$GO_JAVA_HOME/bin/java"
+      fi
+      if [ ! -x "$java_cmd" ] ; then
+          die "ERROR: GO_JAVA_HOME is set to an invalid directory: $GO_JAVA_HOME
+
+Please set the GO_JAVA_HOME variable in your environment to match the
+location of your Java installation."
+      fi
+  else
+      java_cmd="java"
+      which java >/dev/null 2>&1 || die "ERROR: GO_JAVA_HOME is not set and no 'java' command could be found in your PATH.
+
+Please set the GO_JAVA_HOME variable in your environment to match the
+location of your Java installation."
+  fi
+
+  echo "$java_cmd"
+}
+
+[ ! -z $SERVER_MEM ] || SERVER_MEM="512m"
+[ ! -z $SERVER_MAX_MEM ] || SERVER_MAX_MEM="1024m"
+[ ! -z $SERVER_MAX_PERM_GEN ] || SERVER_MAX_PERM_GEN="256m"
+[ ! -z $SERVER_MIN_PERM_GEN ] || SERVER_MIN_PERM_GEN="128m"
+[ ! -z $GO_SERVER_PORT ] || GO_SERVER_PORT="8153"
+[ ! -z $GO_SERVER_SSL_PORT ] || GO_SERVER_SSL_PORT="8154"
+[ ! -z "$SERVER_WORK_DIR" ] || SERVER_WORK_DIR="/var/lib/go-server"
+[ ! -z "$YOURKIT_DISABLE_TRACING" ] || YOURKIT_DISABLE_TRACING=""
+
+if [ -d /logs ]; then
+    STDOUT_LOG_FILE=/logs/go-server.out.log
+else
+    STDOUT_LOG_FILE=go-server.out.log
+fi
+
+if [ -d /var/run/go-server ]; then
+    PID_FILE=/var/run/go-server/go-server.pid
+else
+    PID_FILE=go-server.pid
+fi
+
+GO_CONFIG_DIR=/var/lib/go-server/config
+
+if [ "$JVM_DEBUG" != "" ]; then
+    JVM_DEBUG="-Xdebug -Xrunjdwp:transport=dt_socket,server=y,suspend=n,address=5005"
+else
+    JVM_DEBUG=''
+fi
+
+if [ "$GC_LOG" != "" ]; then
+    GC_LOG="-verbose:gc -Xloggc:go-server-gc.log -XX:+PrintGCTimeStamps -XX:+PrintTenuringDistribution -XX:+PrintGCDetails -XX:+PrintGC"
+else
+    GC_LOG=''
+fi
+
+if [ ! -z $SERVER_LISTEN_HOST ]; then
+    GO_SERVER_SYSTEM_PROPERTIES+="-Dcruise.listen.host=$SERVER_LISTEN_HOST"
+fi
+
+
+# Bootstrapping
+if [ ! -f /config/gocd-server-config.xml ]; then
+  echo "Bootstrapping initial configuration."
+  unzip -qq /go-server/go.jar defaultFiles/config/cruise-config.xml
+  cp defaultFiles/config/cruise-config.xml /config/gocd-server-config.xml
+fi
+AGENT_KEY="${AGENT_KEY:-123456789abcdef}"
+[[ -n "$AGENT_KEY" ]] && sed -i -e 's/agentAutoRegisterKey="[^"]*" *//' -e 's#\(<server\)\(.*artifactsdir.*\)#\1 agentAutoRegisterKey="'$AGENT_KEY'"\2#' /config/gocd-server-config.xml
+if [ ! -d /config/db ]; then
+  echo "Bootstrapping database."
+  mkdir /config/db
+  unzip -qq /go-server/go.jar defaultFiles/h2db.zip defaultFiles/h2deltas.zip
+  unzip -qq defaultFiles/h2db.zip -d /config/db
+  unzip -qq defaultFiles/h2deltas.zip -d /config/db
```

Comment:
If I understand right, `/config/db` can come from a volume. Shouldn't the unzip of h2deltas.zip to `/config/db` happen every time, and not just if `/config/db` does not exist? I mean, it's not part of bootstrapping. If it's not done every time, upgrades will fail. Every release can change h2deltas.zip. So, if you use a new docker image and get volumes from your data-only container, `/config/db` will be bind mounted and the `! -d /config/db` check will be false, and the new deltas for this release won't be in place.

Once this is done: Across releases of GoCD, if we remove a file from h2deltas.zip, the old file will not get removed from `/config/db/h2deltas`, right? This might be a problem. Might want to think about cleaning up the dirs. I suspect this will be a problem with vanilla GoCD as well, and doesn't show up because we only ever add to h2deltas, and never remove from it. So, not very worried about this.